arch/sim/src/sim/up_allocateheap.c

```diff
@@ -71,6 +75,9 @@ static uint8_t sim_heap[SIM_HEAP_SIZE];
 
 void up_allocate_heap(void **heap_start, size_t *heap_size)
 {
+#if defined(CONFIG_LIBC_MODLIB)
```

Comment:
How about elf binary? or let's always host_alloc_heap.

Reply:
elf binaries are out of the scope of this PR. :-)

i don't know if mmap works on windows. i guess not.

Reply:
> elf binaries are out of the scope of this PR. :-)
> 

But, elf binary should has the same problem, why we don't fix together?

> i don't know if mmap works on windows. i guess not.

How about we check BINFMT too?

Reply:
ok. done.